pkg/models/retry.go

```diff
@@ -19,15 +22,21 @@ type Timer interface {
 // RetryIfFunc determines whether a retry should be attempted based on the error.
 type RetryIfFunc func(error) bool
 
+// RetryIfFuncV2 determines whether a retry should be attempted based on the response.
+type RetryIfFuncV2 func(*http.Response, error) bool
```

Comment:
Note: this RetryIfV2 variation gives the user the ability to decide what they want to do based on the original response and the error.  We don't coerce the status into an error prematurely as to give the user more control over the original Response and error to make a better informed decision on when to Retry.